dbt_platform_helper/utils/validation.py

```diff
@@ -456,6 +465,100 @@ def is_between(value):
 )
 
 
+def validate_platform_config(config):
+    get_aws_session_or_abort()  # Ensure we have a valid session as validation requires it.
+    PLATFORM_CONFIG_SCHEMA.validate(config)
+    _validate_environment_pipelines(config)
+    _validate_codebase_pipelines(config)
+
+
+def _validate_environment_pipelines(config):
+    enriched_config = apply_environment_defaults(config)
+    bad_pipelines = {}
+
+    for pipeline_name, pipeline in enriched_config.get("environment_pipelines", {}).items():
+        bad_envs = []
+        account = pipeline.get("account", None)
+        if account:
+            for env in pipeline.get("environments", {}).keys():
+                env_account = (
+                    enriched_config.get("environments", {})
+                    .get(env, {})
+                    .get("accounts", {})
+                    .get("deploy", {})
+                    .get("name")
+                )
+                if env_account and not env_account == account:
+                    bad_envs.append(env)
```

Comment:
Maybe some renaming for clarity and defaulting an account name so the if statement reads clearer?
```suggestion
        pipeline_account = pipeline.get("account", None)
        if pipeline_account:
            for env in pipeline.get("environments", {}).keys():
                env_account = (
                    enriched_config.get("environments", {})
                    .get(env, {})
                    .get("accounts", {})
                    .get("deploy", {})
                    .get("name")
                )
                if not env_account == pipeline_account:
                    bad_envs.append(env)
```